symfony/framework-bundle/3.3/public/index.php

```diff
@@ -2,18 +2,11 @@
 
 use App\Kernel;
 use Symfony\Component\Debug\Debug;
-use Symfony\Component\Dotenv\Dotenv;
 use Symfony\Component\HttpFoundation\Request;
 
 require __DIR__.'/../vendor/autoload.php';
 
-// The check is to ensure we don't use .env in production
-if (!isset($_SERVER['APP_ENV'])) {
-    if (!class_exists(Dotenv::class)) {
-        throw new \RuntimeException('APP_ENV environment variable is not defined. You need to define environment variables for configuration or add "symfony/dotenv" as a Composer dependency to load variables from a .env file.');
-    }
-    (new Dotenv())->load(__DIR__.'/../.env');
-}
+Kernel::bootstrapEnvironment();
```

Comment:
I'm wondering if we should SET `$_SERVER['APP_ENV']` inside `Kernel::bootstrapEnvironment();`. Basically, move this logic there. We would:

1) Load `.env` when needed
2) If `APP_ENV` is still not set, set `$_SERVER['APP_ENV'] = 'dev'`;
3) Except, if `$env` is passed to `bootstrapEnvironment()` (like for console), set that `$env` value into `$_SERVER['APP_ENV']`.

We could do the same for `$_SERVER['APP_DEBUG']`. The idea would be to simplify / de-duplicate all of the `APP_ENV` and `APP_DEBUG` logic. We could even move the `umask` & Debug code into `bootstrapEnvironment()`